### ISSUE TITLE
More granular admin permissions

### ISSUE THREAD

**Issue by GUI** [NREL/api-umbrella#9]
Currently the admin web tool allows admins to access and administer anything and everything. If you have an admin account, you have full access. We need a better permissions model for restricting access to the admin. The main use-case for this is api.data.gov where agency accounts should only be able to view and administer their own agencies data (so I'll use agencies as an example, but this concept should be more generic). The obvious places this extends to are:
- **Analytics:** Agencies should only be able to view their own API's analytics.
- **API Backend:** Agencies should only be able to edit their own API backends. Agencies should be able to add new API backends, but preferably in a way that wouldn't step on any other agencies toes (for example, NREL should not be able to overlap with GSA and define a new API backend with a path like api.data.gov/gsa/nrels-cool-service).
- **API Backend publishing:** Right now publishing API backend changes should maybe be a separate feature that only super-admins have access to, since it has the potential to impact multiple APIs. Ideally, though, an agency could only publish their own backend changes, without pushing any other pending edits from other agencies live.
- **API key management:** API keys are fundamentally meant to be shared across agencies, so this area will have to be shared to some degree. However, there should be restrictions around which roles can be assigned (so I can't assign a role that grants access to a private api belonging to another agency). Disabling API keys completely should be discussed, but I think all admins should still have the ability to disable any api key in the event it's causing abuse. How custom api key rate limits should be handled needs more discussion (the answer may be that we need a way to s

In short, I think we need a concept of admin groups (so multiple users from agencies can share access) and then we need to be able to specify which APIs and roles this group has ownership over. API ownership might involve tying the agency to specific API backends that get configured, or it may involve a broader mechanism for assigning domains and URL paths that the agency "owns." The latter needs a bit more thought to ensure it would work, but it may actually end up being simpler and give us more flexibility. It would also solve the issue of permissions around creating new API backends (since if I own "/nrel/*", then I can create all the new API backends I want under /nrel/whatever, but I would be forbidden from creating /gsa/something-else).
